packages/api-reference/src/v2/blocks/helpers/block-store.ts

```diff
@@ -0,0 +1,75 @@
+import type { AvailableClients } from '@scalar/snippetz'
+import { safeLocalStorage } from '@scalar/helpers/object/local-storage'
+import { LS_CONFIG, debounce } from '@scalar/object-utils/mutator-record'
+import { reactive } from 'vue'
+
+type BlockStoreState = {
+  selectedClient: AvailableClients[number] | null
+}
+
+/** The base global state of the block store */
+const state = reactive<BlockStoreState>({
+  selectedClient: null,
+})
+
+/** Local storage key for the block store */
+export const BLOCK_STORE_LS_KEY = 'scalar-blocks-state'
+
+/** Keeps track of whether the block store has been restored from localStorage */
+let restored = false
```

Comment:
could we use a ref here instead of a let?

Reply:
we don't need a ref here actually